### PR TITLE
NDRS-578: Use smaller subset of chainspec in era supervisor.

### PR DESCRIPTION
**Pull request by goral09**
https://casperlabs.atlassian.net/browse/NDRS-578

### CODE REVIEW

File: node/src/components/consensus/config.rs [hunk 1]
```diff
@@ -11,3 +18,32 @@ pub struct Config {
...
+
+/// Consensus protocol configuration.
+#[derive(DataSize, Debug)]
+pub(crate) struct ProtocolConfig {
+    pub(crate) highway_config: HighwayConfig,
+    /// Number of eras before an auction actually defines the set of validators.
+    /// If you bond with a sufficient bid in era N, you will be a validator in era N +
+    /// auction_delay + 1
+    pub(crate) auction_delay: u64,
+    pub(crate) unbonding_delay: EraId,
```
Comment (afck) [inline node/src/components/consensus/config.rs:30]: That should either be a `u64`, too, or we'd need an `EraIdDiff` type.

File: node/src/components/consensus/config.rs [hunk 2]
```diff
@@ -11,3 +18,32 @@ pub struct Config {
...
+    /// Name of the Genesis network.
```
Comment (afck) [inline node/src/components/consensus/config.rs:31]: Not sure I understand.
```suggestion
    /// Name of the network.
```

File: node/src/components/consensus/config.rs [hunk 3]
```diff
@@ -11,3 +18,32 @@ pub struct Config {
...
+    pub(crate) name: String,
+    /// Timestamp for the Genesis block.
```
Comment (afck) [inline node/src/components/consensus/config.rs:33]: Maybe just "Genesis timestamp", since we don't have a formal genesis block?

**Review by afck — APPROVED** [review]
Looks good! Just a few nit-picks.

File: node/src/components/consensus/config.rs (hunk 1 above)
Comment (goral09) [reply node/src/components/consensus/config.rs:30]: I used the types that were in the original `Chainspec`. Should they be changed there as well?

Comment (afck) [reply node/src/components/consensus/config.rs:30]: I think so. I find it misleading to use `EraId` for a _number_ of eras, i.e. a _difference_ of IDs.

Comment (goral09) [reply node/src/components/consensus/config.rs:30]: Actually, the `EraId` here is just a type alias that is really a `u64`.

File: node/src/components/consensus/config.rs (hunk 3 above)
Comment (goral09) [reply node/src/components/consensus/config.rs]: Done in [`3703796` (#665)](https://github.com/CasperLabs/casper-node/pull/665/commits/3703796ffe51d59e32e7d93af6573e4a1a4a0eea).

File: node/src/components/consensus/config.rs (hunk 2 above)
Comment (goral09) [reply node/src/components/consensus/config.rs]: Done in [`3703796` (#665)](https://github.com/CasperLabs/casper-node/pull/665/commits/3703796ffe51d59e32e7d93af6573e4a1a4a0eea).

File: node/src/components/consensus/config.rs (hunk 1 above)
Comment (goral09) [reply node/src/components/consensus/config.rs]: Done in [`3703796` (#665)](https://github.com/CasperLabs/casper-node/pull/665/commits/3703796ffe51d59e32e7d93af6573e4a1a4a0eea).

**Comment by goral09** [comment]
bors merge

File: node/src/components/consensus/config.rs (hunk 1 above)
Comment (afck) [reply node/src/components/consensus/config.rs]: Thanks! Yes, even if it's just a type alias I'd use `u64` for era ID differences then. :+1:

**Pull request merged**